Quarkus OIDC quarkus.oidc.enabled  and DisabledAuthController doesn't seem to work

**Describe the bug**
Setting this `quarkus.oidc.enabled=false` doesnt "disable" the OIDC features. Added the below code as well
```
import javax.annotation.Priority;
import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.inject.Alternative;
import javax.interceptor.Interceptor;

import io.quarkus.security.spi.runtime.AuthorizationController;

@Alternative
@Priority(Interceptor.Priority.LIBRARY_AFTER)
@ApplicationScoped
public class DisabledAuthController extends AuthorizationController{

	
    @Override
    public boolean isAuthorizationEnabled() {
        return false;
    }

}
This results in getting a Runtime Exception for ambiguous dependency.

`[ERROR] org.acme.security.keycloak.authorization.CheckTest.testAgentGet  Time elapsed: 0.009 s  <<< ERROR!                                                              java.lang.RuntimeException:                                                                                                                                             java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors                                                                        [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: Found 3 deployment problems:                                                                                                                                                                   [1] Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                           - java member: io.quarkus.vertx.http.runtime.security.HttpAuthorizer#controller                                                                                         - declared on CLASS bean [types=[io.quarkus.vertx.http.runtime.security.HttpAuthorizer, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.vertx.http.runtime.security.HttpAuthorizer]                                                                                                                                           - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                   [2] Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                           - java member: io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor#controller                                                                               - declared on INTERCEPTOR bean [bindings=[@RolesAllowed(value = [""])], target=Optional[io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor]]               - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                   [3] Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                           - java member: io.quarkus.security.runtime.interceptor.AuthenticatedInterceptor#controller                                                                              - declared on INTERCEPTOR bean [bindings=[@Authenticated], target=Optional[io.quarkus.security.runtime.interceptor.AuthenticatedInterceptor]]                           - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                           at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1002)                                                                                      at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:234)                                                                                                at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:122)                                                                                            at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:391)                                                                                               at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                                                                       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                                                                     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)                                                             at java.base/java.lang.reflect.Method.invoke(Method.java:566)                                                                                                           at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:936)                                                                                            at io.quarkus.builder.BuildContext.run(BuildContext.java:277)                                                                                                           at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)                                                                     at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)                                                                                     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)                                                                        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)                                                                              at java.base/java.lang.Thread.run(Thread.java:834)                                                                                                                      at org.jboss.threads.JBossThread.run(JBossThread.java:479)                                                                                                              Suppressed: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                                                                                                                             - java member: io.quarkus.vertx.http.runtime.security.HttpAuthorizer#controller                                                                                         - declared on CLASS bean [types=[io.quarkus.vertx.http.runtime.security.HttpAuthorizer, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.vertx.http.runtime.security.HttpAuthorizer]                                                                                                                                           - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                                   at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:513)                                                                                                 at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:362)                                                                                                            at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:226)                                                                                                ... 14 more                                                                                                                                                     Suppressed: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                                                                                                                             - java member: io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor#controller                                                                               - declared on INTERCEPTOR bean [bindings=[@RolesAllowed(value = [""])], target=Optional[io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor]]               - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                                   at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:513)                                                                                                 at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:362)                                                                                                            at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:232)                                                                                                ... 14 more                                                                                                                                                     Suppressed: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                                                                                                                             - java member: io.quarkus.security.runtime.interceptor.AuthenticatedInterceptor#controller                                                                              - declared on INTERCEPTOR bean [bindings=[@Authenticated], target=Optional[io.quarkus.security.runtime.interceptor.AuthenticatedInterceptor]]                           - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                                   at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:513)                                                                                                 at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:362)                                                                                                            at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:232)                                                                                                ... 14 more                                                                                                                                                                                                                                                                                                                     Caused by: java.lang.RuntimeException:                                                                                                                                  io.quarkus.builder.BuildException: Build failure: Build failed due to errors                                                                                                    [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: Found 3 deployment problems:                                                                                                                                                                   [1] Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                           - java member: io.quarkus.vertx.http.runtime.security.HttpAuthorizer#controller                                                                                         - declared on CLASS bean [types=[io.quarkus.vertx.http.runtime.security.HttpAuthorizer, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.vertx.http.runtime.security.HttpAuthorizer]                                                                                                                                           - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                   [2] Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                           - java member: io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor#controller                                                                               - declared on INTERCEPTOR bean [bindings=[@RolesAllowed(value = [""])], target=Optional[io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor]]               - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]                                                                          - CLASS bean [types=[io.quarkus.test.security.TestAuthController, io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.test.security.TestAuthController]                                                                                                                  - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.security.spi.runtime.AuthorizationController]                                                                                                                                   [3] Ambiguous dependencies for type io.quarkus.security.spi.runtime.AuthorizationController and qualifiers [@Default]                                                           - java member: io.quarkus.security.runtime.interceptor.AuthenticatedInterceptor#controller                                                                              - declared on INTERCEPTOR bean [bindings=[@Authenticated], target=Optional[io.quarkus.security.runtime.interceptor.AuthenticatedInterceptor]]                           - available beans:                                                                                                                                                              - CLASS bean [types=[io.quarkus.security.spi.runtime.AuthorizationController, org.acme.security.keycloak.authorization.DisabledAuthController, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.security.keycloak.authorization.DisabledAuthController]  `


```I need to comment out my annotations about "@RolesAllowed" to disable oauth verification.


**Expected behavior**
Disable security "checks". I do not wish to comment out code lines like @RolesAllowed. I should be able to get a valid response when the security checks are turned off for the project and not  403 forbidden. 

**Actual behavior**
I initially did OIDC quarkus.oidc.enabled =false and it gave me 403 forbidden for the endpoints at runtime. When I read further I added the DisabledAuthController to override it and subsequently started facing the bean issue. 


**To Reproduce**

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).

Or attach an archive containing the reproducer to the issue.

Steps to reproduce the behavior:
Using the quickstart projects, "security-keycloak-authorization-quickstart" with the following changes
1. Set this property: quarkus.oauth2.enabled=false
 2. Add a @RolesAllowed("anything") on a route.
3. Create a Test for this route and annotate it with @TestSecurity(authorizationEnabled = false)
4. Write the below code as well
```
import javax.annotation.Priority;
import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.inject.Alternative;
import javax.interceptor.Interceptor;

import io.quarkus.security.spi.runtime.AuthorizationController;

@Alternative
@Priority(Interceptor.Priority.LIBRARY_AFTER)
@ApplicationScoped
public class DisabledAuthController extends AuthorizationController{

	
    @Override
    public boolean isAuthorizationEnabled() {
        return false;
    }

}




**Configuration**
```properties
# Add your application.properties here, if applicable.

quarkus.oidc.auth-server-url=http://localhost:8180/auth/realms/quarkus
quarkus.oidc.client-id=backend-service
quarkus.oidc.credentials.secret=secret
quarkus.http.cors=true
%test.quarkus.oidc.enabled=false
%dev.quarkus.oidc.enabled=false
# Enable Policy Enforcement
quarkus.keycloak.policy-enforcer.enable=false

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java):  11
 - Quarkus version or git rev:  1.8.3.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
(Add any other context about the problem here.)

--- a/docs/src/main/asciidoc/security-customization.adoc
+++ b/docs/src/main/asciidoc/security-customization.adoc
@@ -236,7 +236,7 @@ public class DisabledAuthController extends AuthorizationController {
 }
 ----
 
-Please also see link:security-testing#testing-security[TestingSecurity Annotation] section how to disable the security checks using `TestSecurity` annotation.
+Please also see link:security-testing#testing-security[TestingSecurity Annotation] section on how to disable the security checks using `TestSecurity` annotation.
 
 == Registering Security Providers
 

--- a/docs/src/main/asciidoc/security-customization.adoc
+++ b/docs/src/main/asciidoc/security-customization.adoc
@@ -236,6 +236,8 @@ public class DisabledAuthController extends AuthorizationController {
 }
 ----
 
+Please also see link:security-testing#testing-security[TestingSecurity Annotation] section how to disable the security checks using `TestSecurity` annotation.
+
 == Registering Security Providers
 
 === Default providers

